README.md

```diff
@@ -0,0 +1,262 @@
+| Authorization      |     | Authorization    |                            
+| Manager            |---->| SPI              |                            
++--------------------+     +------------------+                            
+     |                                                                     
+     v                                                 +------------------+
++--------------------+      +--------------------+     | Cryptography     |
+| Core Logic         |----->| Cryptography       |---->| SPI              |
++--------------------+      +--------------------+     +------------------+
+     |                                                                     
+     v                                                                     
++--------------------+     +------------------+                            
+| Storage            |     | Repository       |                            
+| Repositories       |---->| SPIs             |                            
++--------------------+     +------------------+                            
+```
+
+### Clients SDK
+The Java SDK provides a robust, production-ready client with:
+
+- **Automatic Retries** - Exponential backoff for transient failures
+- **Connection Pooling** - Efficient HTTP connection management
+- **Metrics Integration** - Micrometer and JMX support
+- **Structured Logging** - MDC context for request tracing
+
+See the [detailed client documentation](clients/java/README.md) for advanced usage.
+
+Clients are provided as separate modules in the `clients` folder. For now only Java clients are provided, with support for Java 8 and higher.
+```text
+clients/              # Client libraries
+└── java/             # Java SDK
+    ├── client-api/   # Public interfaces
+    └── client-impl/  # Implementation
+```
+
+## Configuration
+
+### Using own implementation for authentication and authorization
+Create a java application similar to `simple-app` and add the server's dependency in `pom.xml`.
+```xml
+<dependency>
+    <groupId>com.flipkart.grayskull</groupId>
+    <artifactId>server</artifactId>
+    <version>0.1.0</version>
+</dependency>
+```
+Optionally add own implementations of `GrayskullAuthenticationProvider` and `GrayskullAuthorizationProvider` in the application.  
+If your implementations are in a separate jar like `derby-async-audit` then make sure that spring boot scans them first like below
+assuming your SPI implementations are in `com.example.spi.impl` and application is in `com.example.app`
+```java
+@SpringBootApplication(scanBasePackages = {"com.example.spi.impl", "com.flipkart.app"})
+public class GrayskullApplication {
+    public static void main(String[] args) {
+        SpringApplication.run(GrayskullApplication.class, args);
+    }
+}
+```
+This way own SPI implementations are picked up first instead of simple implementations provided in `server` and used by the server.  
```

Comment:
provide a couple more examples of SPI impl usage. Say, a custom audit module, or a custom authn module.